### PR TITLE
[fixed] incorrect focusAfterRender on componentDidUpdate.

### LINKED ISSUE #315: Undefined error in contentHasFocus method
### Summary:
Got the following error in the browser console when using react modal
Uncaught TypeError: Cannot read property 'contains' of undefined
    at Object.contentHasFocus (ModalPortal.js:171)
    at Object.focusContent (ModalPortal.js:101)
    at Object.componentDidUpdate (ModalPortal.js:66)
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponent.js:729
    at CallbackQueue.notifyAll (CallbackQueue.js:76)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:80)
    at ReactReconcileTransaction.closeAll (Transaction.js:206)
    at ReactReconcileTransaction.perform (Transaction.js:153)
    at ReactUpdatesFlushTransaction.perform (Transaction.js:140)
### Steps to reproduce:

 1. When populating contents in the modal dynamically
 2.
 3.

### Expected behavior:
Should be able to populate contents without any errors.
### Additional notes:
A small undefined check in the react-modal.js resolved the issue. Will make a PR with the fix.

### PR DESCRIPTION
**Pull request by lindskogen**
Maybe fixes #315.

Issue #315 seems to be a problem with inconsistent state when using react-modal with `redux` [1] (all errors seems to be reported by developer using redux).

It's not a proper fix since this behaviour was not reproduced, but it should be safe.

[1] - maybe a redux and/or component state async update. (?)

Changes proposed:
- don't try to focus if `this.content` is not available.
- Uses bound ref functions instead of arrow functions, as suggested by https://facebook.github.io/react/docs/refs-and-the-dom.html#caveats

Acceptance Checklist:
- [x] All commits have been squashed to one.
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [ ] If this is a code change, a spec testing the functionality has been added.
- [ ] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

**Comment by diasbruno** [comment]
Hi @lindskogen, thank you. There are a few lint problems. You can use `npm run lint` to see what is going on.

**Comment by diasbruno** [comment]
@lindskogen One last thing, can you rewrite the commit message to start with `[fixed]`? This helpful to generate the changelog.

**Comment by lindskogen** [comment]
@diasbruno I think that did it!

**Comment by diasbruno** [comment]
I was trying to reproduce this issue (#315) and I think got it. 
I've added some information on the issue.

Unfortunately, this won't fix the issue, but it will be merged as a recommendation/good practice.
Thank you for having a look on this.

File: src/components/ModalPortal.js [hunk 1]
```diff
@@ -122,7 +130,8 @@ export default class ModalPortal extends Component {
   }
 
   // Don't steal focus from inner elements
-  focusContent = () => (!this.contentHasFocus()) && this.content.focus();
+  focusContent = () =>
```
Comment (diasbruno) [inline src/components/ModalPortal.js]: Sorry, I've forgot to remote this trailing space after `=> `.

**Comment by diasbruno** [comment]
@lindskogen thank you so much!

**Comment by diasbruno** [comment]
Released `v2.0.6`.

**Pull request merged**